src/rgw/driver/sfs/sqlite/sqlite_buckets.cc

```diff
@@ -97,4 +97,21 @@ std::vector<std::string> SQLiteBuckets::get_deleted_buckets_ids() const {
   );
 }
 
+bool SQLiteBuckets::bucket_empty(const std::string& bucket_id) const {
+  auto storage = conn->get_storage();
+  // Finds versions that are committed and regular (no delete markers)
+  auto ids = storage.select(
+      &DBVersionedObject::id,
+      inner_join<DBObject>(
+          on(is_equal(&DBObject::uuid, &DBVersionedObject::object_id))
+      ),
+      where(
+          is_equal(&DBVersionedObject::object_state, ObjectState::COMMITTED) and
+          is_equal(&DBObject::bucket_id, bucket_id) and
+          is_equal(&DBVersionedObject::version_type, VersionType::REGULAR)
+      )
+  );
+  return ids.size() == 0;
```

Comment:
This transfers all ids from SQLite to a std::vector just take the count. Suggest to take the count in SQLite or return a booleanesque value to return a constant size result.

Example:
sqlite> select count(id) > 0 from versioned_objects inner join objects on (uuid = object_id) where object_state == 1 and version_type == 0;

Reply:
I changed to a plain `count` with `inner join` instead of a `select count...` 
That way it returns just the number I'm looking for instead or a vector of ints.